packages/react/src/auto/mui/inputs/MUIAutoInput.tsx

```diff
@@ -19,36 +17,39 @@ const FieldTypeToInputType: Partial<Record<GadgetFieldType, TextFieldProps["type
   [FieldType.EncryptedString]: "password",
 };
 
-export const MUIAutoFormControl = (props: { path: string; control: Control<any>; field: FieldMetadata; children: ReactElement }) => {
+export const MUIAutoFormControl = (props: { field: string; children: ReactElement }) => {
+  const { path, metadata: fieldMetadata } = useFieldMetadata(props.field);
   const {
     fieldState: { error },
   } = useController({
-    name: props.path,
-    control: props.control,
-    rules: { required: props.field.requiredArgumentForInput },
+    name: path,
+    rules: { required: fieldMetadata.requiredArgumentForInput },
   });
 
   return (
-    <FormControl {...props.field} error={!!error}>
+    <FormControl {...fieldMetadata} error={!!error}>
       <FormGroup>
-        <FormControlLabel label={props.field.name} control={props.children} />
+        <FormControlLabel label={fieldMetadata.name} control={props.children} />
       </FormGroup>
       {error && <FormHelperText>{error?.message}</FormHelperText>}
     </FormControl>
   );
 };
 
-export const MUIFormInput = (props: { path: string; field: FieldMetadata; control: Control<any> }) => {
+export const MUIAutoInput = (props: { field: string }) => {
+  const { path, metadata: fieldMetadata } = useFieldMetadata(props.field);
```

Comment:
Maybe we can just call this field metadata -- I think it's clear enough that it is the metadata for the field, and we can resolve the field/fieldMetadata ambiguity?

Reply:
Updated so that this is just `metadata` for all places that use this hook